src/targeting.js

```diff
@@ -350,6 +350,7 @@ export function newTargeting(auctionManager) {
    * bid caching done with all bids specifically for Marfeel purposes due to its own configuration
    */
   const filterBidsByAdUnit = (bidsReceived) => bidsReceived.filter(bid => latestAuctionForAdUnit[bid.adUnitCode] === bid.auctionId);
+  const isBidAlreadyRecieved = (bid, lastLocation) => bidsByReferrer[lastLocation] && !!bidsByReferrer[lastLocation].find(bidCached => bidCached.adId === bid.adId);
```

Comment:
This shouldn't be `!!bidsByReferrer[lastLocation] && bidsByReferrer[lastLocation].find(bidCached => bidCached.adId === bid.adId);`